src/azul/plugins/__init__.py

```diff
@@ -331,6 +332,19 @@ def exposed_indices(self) -> Mapping[str, Sorting]:
         """
         raise NotImplementedError
 
+    @property
+    @abstractmethod
+    def page_size_limits(self) -> Mapping[EntityType, int]:
+        """
+        Some types of outer entity aggregates can reach very large sizes,
+        causing timeouts or other errors when requests are made to their
+        associated index using large page sizes. This mapping determines the
+        maximum page size for each exposed index, allowing us to set lower
+        limits for larger entities while preserving the higher limits for
+        smaller ones.
```

Comment:
A bit verbose. Entity sizes differ between catalogs and entity types, so it's kind of obvious why one would scale the page size limits with them.